### PR TITLE
Flink: Update to KUDO version 0.3.0

### PR DESCRIPTION
**Pull request by fabianbaier**
**What type of PR is this?**
/kind cleanup
/kind enhancement

**What this PR does / why we need it**:
- Adding a new minor version to Flink (`0.2.0`).
- Refactoring Flink demo to work with new KUDO `0.2.0` (don't confuse the independent SemVer 2, they just coincide in both versions with `0.2.0`)
- Adding documentation around how to run `financial fraud demo`

### CODE REVIEW

File: repository/flink/docs/demo/financial-fraud/README.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+This demo implements a data processing infrastructure with KUDO that is able to spot money laundering. In the context of money laundering, we  want to detect amounts larger than $10.000 transferred between two accounts, even if that amount is split into many small batches.  See also [US](https://www.fincen.gov/history-anti-money-laundering-laws) and [EU](http://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX%3A32015L0849) legislation and regulations on this topic for more information.
+
+The architecture follows more or less the [SMACK stack architecture](https://mesosphere.com/blog/smack-stack-new-lamp-stack/):
+- Events: Event are being generated by a small [GOLANG generator](https://github.com/dcos/demos/blob/master/flink/1.11/generator/generator.go). The events are in the form 'Sunday, 23-Jul-17 01:06:47 UTC;66;26;7810', where the first field '23-Jul-17 01:06:47 UTC' represents the (increasing) timestamp of transactions; the second field '66' represent the sender account; the third field the receiver account; and the fourth field represent the dollar amount transferred during that transaction.
```
Comment (gkleiman) [inline repository/flink/docs/demo/financial-fraud/README.md]:
```suggestion
- Events: Event are being generated by a small [Golang generator](https://github.com/dcos/demos/blob/master/flink/1.11/generator/generator.go). The events are in the form 'Sunday, 23-Jul-17 01:06:47 UTC;66;26;7810', where the first field '23-Jul-17 01:06:47 UTC' represents the (increasing) timestamp of transactions; the second field '66' represent the sender account; the third field the receiver account; and the fourth field represent the dollar amount transferred during that transaction.
```

File: repository/flink/docs/demo/financial-fraud/README.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+- Ingestion: The generated events are being ingested and buffered by a Kafka queue with the default topic 'transactions'. Being a Microservice we will deploy the data-generator on kubernetes.
+- Stream Processing: As we require fast response times, we use Apache Flink as a Stream processor running the [FinancialTransactionJob](https://github.com/dcos/demos/tree/master/flink/1.10/flink-job/src/main/java/io/dcos).
+- Storage: Here we diverge a bit from the typical SMACK stack setup and don't write the results into a Datastore such as Apache Cassandra. Instead we write the results again into a Kafka Stream (default: 'fraud'). Note, that Kafka also offers data persistence for all unprocessed events.
+- Actor: In order to view the results we use again a small [Golang viewer](https://github.com/dcos/demos/blob/master/flink/1.11/actor/actor_viewer.go) which simply reads and displays the results from the output Kafka stream. Being a Microservice we will deploy the viewer on kubernetes.
+
+## Prerequisites
+
+Before you get started:
+
+- Have KUDO run in your cluster [ [Getting Started](https://kudo.dev/docs/getting-started/) ]
```
Comment (gkleiman) [inline repository/flink/docs/demo/financial-fraud/README.md]:
```suggestion
- Have KUDO installed on your cluster [ [Getting Started](https://kudo.dev/docs/getting-started/) ]
```

File: repository/flink/docs/demo/financial-fraud/README.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+- Have the `zookeeper` Framework with `0.1.0` as FrameworkVersion installed 
+    - Use the KUDO CLI with the following command:
+        ```bash
+        $ kubectl kudo install zookeeper --package-version=0.1.0
+        framework.kudo.k8s.io/v1alpha1/zookeeper created
+        frameworkversion.kudo.k8s.io/v1alpha1/zookeeper-0.1.0 created
+        No Instance tied to this "zookeeper" version has been found. Do you want to create one? (Yes/no) no
+
+        ```
+        *Note: Don't forget to say "**no**" when prompted, we just want to install the Framework and FrameworkVersion objects but not instantiated an Instance object*
```
Comment (gkleiman) [inline repository/flink/docs/demo/financial-fraud/README.md]:
```suggestion
        *Note: Don't forget to say "**no**" when prompted, we just want to install the Framework and FrameworkVersion objects but we don't want to instantiate an Instance object*
```

File: repository/flink/docs/demo/financial-fraud/README.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+- Have the `kafka` Framework with `0.1.0` as FrameworkVersion installed 
+    - Use the KUDO CLI with the following command:
+        ```bash
+        $ kubectl kudo install kafka --package-version=0.1.0
+        framework.kudo.k8s.io/v1alpha1/kafka created
+        frameworkversion.kudo.k8s.io/v1alpha1/kafka-0.1.0 created
+        No Instance tied to this "kafka" version has been found. Do you want to create one? (Yes/no) no
+
+        ```
+        *Note: Don't forget to say "**no**" when prompted, we just want to install the Framework and FrameworkVersion objects but not instantiated an Instance object*
```
Comment (gkleiman) [inline repository/flink/docs/demo/financial-fraud/README.md]:
```suggestion
        *Note: Don't forget to say "**no**" when prompted, we just want to install the Framework and FrameworkVersion objects but we don't want to instantiate an Instance object*
```

File: repository/flink/docs/demo/financial-fraud/README.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+SUBMITTED JOB!
+```
+
+To get the fraud output from the actor:
+
+```bash
+kubectl logs $(kubectl get pod -l step=act -o jsonpath="{.items[0].metadata.name}")
+```
+
+Congratulations, you just installed an high available Flink cluster that runs a financial fraud detection job!
```
Comment (gkleiman) [inline repository/flink/docs/demo/financial-fraud/README.md]:
```suggestion
Congratulations, you just installed a highly available Flink cluster that runs a financial fraud detection job!
```

**Comment by runyontr** [comment]
It's hard to tell what's actually different between Flink 0.1.0 and Flink 0.2.0.  

I like the documentation around the demo

File: repository/flink/docs/demo/financial-fraud/README.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
+# Flink Financial Fraud Demo
+
+This demo follows the outline provided by [DCOS's](https://github.com/dcos/demos/tree/master/flink-k8s/1.11) demo
```
Comment (yankcrime) [inline repository/flink/docs/demo/financial-fraud/README.md:3]: There is an updated version of this demo whose URL is https://github.com/dcos/demos/tree/master/flink-k8s/1.13

Comment (fabianbaier) [reply repository/flink/docs/demo/financial-fraud/README.md:3]: Looking over https://github.com/dcos/demos/pull/86 it seems there are overall minor wording changes. Was there anything in particular note worthy that we should add in this demo here as well?

Comment (yankcrime) [reply repository/flink/docs/demo/financial-fraud/README.md:3]: I think @gkleiman caught most of - and made similar recommendations to - anything wording related.

**Comment by fabianbaier** [comment]
the basic Flink operator works, however the financial fraud demo is right now making some troubles after https://github.com/kudobuilder/kudo/pull/447/files#diff-c2e52ee1aa92f961c976702674f98591R500  ( see https://kubernetes.slack.com/archives/CG3HTFCMV/p1561702908340600 )

**Review by jbarrick-mesosphere — APPROVED** [review]

**Pull request merged**